src/components/adaptive-cards/Component/Component.jsx

```diff
@@ -118,6 +120,10 @@ export default function Component({data, className, style: styleProp}) {
         break;
       case acPropTypes.action:
       case acPropTypes.children:
+        cc = value.flatMap((child) => [{type: 'Spacer', separator: child.separator, spacing: child.spacing || 'default'}, child]);
+
+        acData[prop] = cc;
```

Comment:
this actually modifies the received `data` object (you are sending `data` and not `acData` to `C` on line 201 below; they actually reference the same object in memory). This means that, at the next render, a new set of Spacers will be added to the list of children already containing Spacers.

One solution is to create a copy of the data object:

```js
let compData = data;

...

case acPropTypes.children:
  const childrenWithSpacers = value.flatMap(...);
  compData = {...compData, [prop]: childrenWithSpacers};
  break;

...

return <C data={compData} ... >;
```